pkg/backend/httpstate/state.go

```diff
@@ -154,21 +152,20 @@ func (u *cloudUpdate) recordEvent(
 	// Send the event to the Pulumi Service to power things like the update summary page.
 	// Currently opt-in via flag to allow for gathering per data before the service-side changes
 	// are available in production.
-	if cmdutil.IsTruthy(os.Getenv("PULUMI_RECORD_ENGINE_EVENTS")) {
-		apiEvent, convErr := convertEngineEvent(event)
-		if convErr != nil {
-			return errors.Wrap(convErr, "converting engine event")
-		}
-		apiEvent.Sequence = sequenceNumber
-		apiEvent.Timestamp = int(time.Now().Unix())
-		if err = u.backend.client.RecordEngineEvent(u.context, u.update, apiEvent, token); err != nil {
-			return err
-		}
+	apiEvent, convErr := convertEngineEvent(event)
+	if convErr != nil {
+		return errors.Wrap(convErr, "converting engine event")
+	}
+	apiEvent.Sequence = sequenceNumber
+	apiEvent.Timestamp = int(time.Now().Unix())
+	if err = u.backend.client.RecordEngineEvent(u.context, u.update, apiEvent, token); err != nil {
+		return err
 	}
 
 	// We also pre-render the event using the DiffView and post as applicable. Ideally this data
 	// is redundant because we could produce the same log from the raw engine event stream, which
 	// is stored above.
+	// TODO[pulumi-service/issues/2549]: Obviate the need for this.
```

Comment:
Let's not add links to private issues in the public code base. Can you file a sister issue in pulumi/pulumi?